File: libs/jgdsl/src/gdsl/rreil/Address.java
package gdsl.rreil;

import gdsl.rreil.IAddress;
import gdsl.rreil.linear.LinearExpression;

public class Address implements IAddress {
	protected long size;

	public long getSize() {
		return size;
	}

	protected LinearExpression address;

	public LinearExpression getAddress() {
		return address;
	}

	public Address(long size, LinearExpression address) {
		this.size = size;
		this.address = address;
	}
	
	@Override
	public String toString() {
		return address + ":" + size;
	}
}

File: libs/cppgdsl/src/rreil/address.cpp
#include "cppgdsl/rreil/address.h"

#include <iostream>
#include <sstream>

namespace gdsl {
namespace rreil {

address::address(int_t size, std::unique_ptr<linear> lin)
    : size(size), lin(move(lin)) {}

std::string address::to_string() const {
  std::stringstream o;
  o << *this;
  return o.str();
}

std::ostream& operator<<(std::ostream& out, address const& _this) {
  out << "(" << *_this.lin << "/" << _this.size << ")";
  return out;
}

std::unique_ptr<address> address::copy() const {
  return std::unique_ptr<address>(new address(*this));
}

void address::accept(visitor& v) const { v.visit(this); }

}  // namespace rreil
}  // namespace gdsl
